pypesto/objective/roadrunner/petab_importer_roadrunner.py

```diff
@@ -0,0 +1,371 @@
+    form of the likelihood depends on the noise model specified in the provided PEtab problem.
+    For more information, see the
+    `PEtab documentation <https://petab.readthedocs.io/en/latest/documentation_data_format.html#noise-distributions>`_.
+    """  # noqa
+
+    def __init__(
+        self, petab_problem: petab.Problem, validate_petab: bool = True
+    ):
+        """Initialize importer.
+
+        Parameters
+        ----------
+        petab_problem:
+            Managing access to the model and data.
+        validate_petab:
+            Flag indicating if the PEtab problem shall be validated.
+        """
+        self.petab_problem = petab_problem
+        self.rr = roadrunner.RoadRunner()
+
+        self.validate_petab = validate_petab
+        if self.validate_petab:
+            if petab.lint_problem(petab_problem):
+                raise ValueError("Invalid PEtab problem.")
+
+    @staticmethod
+    def from_yaml(yaml_config: Union[Path, str]) -> PetabImporterRR:
+        """Simplified constructor using a petab yaml file."""
+        petab_problem = petab.Problem.from_yaml(yaml_config)
+
+        return PetabImporterRR(petab_problem=petab_problem)
+
+    def _check_noise_formulae(
+        self,
+        edatas: Optional[list[ExpData]] = None,
+        parameter_mapping: Optional[list[ParMappingDictQuadruple]] = None,
+    ):
+        """Check if the noise formulae are valid.
+
+        Currently, only static values or singular parameters are supported.
+        Complex formulae are not supported.
+        """
+        # check that parameter mapping is available
+        if parameter_mapping is None:
+            parameter_mapping = self.create_parameter_mapping()
+        # check that edatas are available
+        if edatas is None:
+            edatas = self.create_edatas()
+        # check that noise formulae are valid
+        for edata, par_map in zip(edatas, parameter_mapping):
+            for noise_formula in edata.noise_formulae:
+                # constant values are allowed
+                if isinstance(noise_formula, numbers.Number):
+                    continue
+                # single parameters are allowed
+                if noise_formula in par_map[1].keys():
+                    continue
+                raise NotImplementedError(
+                    "Noise formulae must be either constants or single "
+                    "parameters. For more complex noise models, please "
+                    "use amici for now."
+                )
+
+    def _write_observables_to_model(self):
+        """Write observables of petab problem to the model."""
+        # add all observables as species
+        for obs_id in self.petab_problem.observable_df.index:
+            try:
+                self.rr.addParameter(obs_id, 0.0)
+            except RuntimeError:
+                warnings.warn(
+                    "Observable already exists in model. Skipping.",
+                    stacklevel=2,
+                )
```

Comment:
The idea was the following:
Petablint has already been run, therefore the petab problem is "well posed" and any parameter that occurs in the observable table has not been in the sbml model. Thus if it now appears in the roadrunner model it may be due to the following two things:
* Someone manually added these parameters
* Someone already ran the `_write_observables_to_model`

I would say in both cases they should be notified of this but the creation of the parameters is fine, and the assignment rules for observables will get overwritten.

Does this reasoning make sense? Or why specifically would you raise an error here?

Reply:
> I would say in both cases they should be notified 

I wouldn't want to support either of those cases. If the model was modified externally between validation and calling this function, it's in an undefined state and a lot of things can go wrong.

Reply:
ok, just removed the try..except. Makes sense to call an error here I guess with your reasoning. 👍🏼